proof_test.go

```diff
@@ -3,14 +3,58 @@ package nmt
 import (
 	"bytes"
 	"crypto/sha256"
+	"io"
 	"testing"
 
 	"github.com/celestiaorg/merkletree"
```

Comment:
Ideally, we would remove this dependency entirely. But orthogonal to this PR.

Reply:
Done

Reply:
Edit -- This cannot be removed, merkletree relies on io errors internally, which is wild!

Reply:
Anyway, this is only in a test, I think should be fine?